packages/office-ui-fabric-react/src/components/FocusZone/FocusZone.tsx

```diff
@@ -924,6 +935,150 @@ export class FocusZone extends React.Component<IFocusZoneProps> implements IFocu
+        return this.focusElement(getNextElement(
+          this._root.current,
+          this._root.current.firstElementChild as HTMLElement,
+          true
+        ) as HTMLElement);
+      }
+    }
+    return changedFocus;
+  }
+  private _movePageDown(): boolean {
+    const leftAlignment = this._focusAlignment.x;
+    if (
+      this._moveFocusPaging(true, (activeRect: ClientRect, targetRect: ClientRect) => {
+        let distance = -1;
+        // ClientRect values can be floats that differ by very small fractions of a decimal.
+        // If the difference between top and bottom are within a pixel then we should treat
+        // them as equivalent by using Math.floor. For instance 5.2222 and 5.222221 should be equivalent,
+        // but without Math.Floor they will be handled incorrectly.
+        const targetRectTop = Math.floor(targetRect.top);
+        const activeRectBottom = Math.floor(activeRect.bottom);
+        if (targetRectTop < activeRectBottom) {
+          if (!this._shouldWrapFocus(this._activeElement as HTMLElement, NO_VERTICAL_WRAP)) {
+            return LARGE_NEGATIVE_DISTANCE_FROM_CENTER;
+          }
+          return LARGE_DISTANCE_FROM_CENTER;
+        } else {
+          if (leftAlignment >= targetRect.left && leftAlignment <= targetRect.left + targetRect.width) {
+            distance = 0;
+          } else {
+            distance = Math.abs(targetRect.left + targetRect.width / 2 - leftAlignment);
+          }
+        }
+        return distance;
+      })
+    ) {
+      this._setFocusAlignment(this._activeElement as HTMLElement, false, true);
+      return true;
+    }
+    return false;
+  }
+
+  private _movePageUp(): boolean {
+    const leftAlignment = this._focusAlignment.x;
+    if (
+      this._moveFocusPaging(false, (activeRect: ClientRect, targetRect: ClientRect) => {
+        let distance = -1;
+        // ClientRect values can be floats that differ by very small fractions of a decimal.
+        // If the difference between top and bottom are within a pixel then we should treat
+        // them as equivalent by using Math.floor. For instance 5.2222 and 5.222221 should be equivalent,
+        // but without Math.Floor they will be handled incorrectly.
+        const targetRectBottom = Math.floor(targetRect.bottom);
+        const activeRectTop = Math.floor(activeRect.top);
+
+        if (targetRectBottom > activeRectTop) {
+          if (!this._shouldWrapFocus(this._activeElement as HTMLElement, NO_VERTICAL_WRAP)) {
+            return LARGE_NEGATIVE_DISTANCE_FROM_CENTER;
+          }
+          return LARGE_DISTANCE_FROM_CENTER;
+        } else {
+          if (leftAlignment >= targetRect.left && leftAlignment <= targetRect.left + targetRect.width) {
+            distance = 0;
+          } else {
+            distance = Math.abs(targetRect.left + targetRect.width / 2 - leftAlignment);
+          }
+        }
+        return distance;
+      })
```

Comment:
This function looks almost exactly like the one in `_movePageDown` (outside of `targetRectTop < activeRectBottom` vs `targetRectBottom > activeRectTop`. It'd be great if this was pulled out into a helper function that took a function for this `if` check, and returned a function (e.g `(activeRect: ClientRect, targetRect: ClientRect) => number`) to pass here... You can also pull that signature into a type to make it easier to reason about, like: `type GetDistanceFromCenter = (activeRect: ClientRect, targetRect: ClientRect) => number;`